src/main/java/duke/ui/Ui.java

```diff
@@ -0,0 +1,38 @@
+package duke.ui;
+
+import java.util.Scanner;
+
+public class Ui {
+    protected Scanner sc;
+
+    public Ui() {
+        this.sc = new Scanner(System.in);
+    }
+
+    public void showWelcome() {
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println("Hello from\n" + logo);
+        System.out.println("Hello! I'm Duke");
+        System.out.println("What can I do for you?");
+    }
+
+    public String readCommand() {
+        return sc.nextLine();
+    }
+
+    public void showLine() {
+        System.out.println("_______________________");
+    }
+
+    public void showError(String err) {
+        System.out.println(err);
+    }
+
+    public void show(String msg) {
+        System.out.println(msg);
+    }
```

Comment:
Perhaps an additional class could be used to manage the logic of printing output by the various Commands. Instead of having to use the `show(String msg)` function each time an output line is displayed.